lucetc/src/function.rs

```diff
@@ -302,4 +306,136 @@ impl<'a> FuncEnvironment for FuncInfo<'a> {
         let inst = pos.ins().call(mem_size_func, &[vmctx]);
         Ok(*pos.func.dfg.inst_results(inst).first().unwrap())
     }
+
+    fn before_translate_operator(
```

Comment:
This entire function is one of the two parts of the `cranelift-wasm` side of this change - adding `before_translate_operator` and `after_translate_operator` functions to `FuncEnvironment` to support cranelift users prodding the instruction stream as it's translated. That's a separate PR and will have to land before this one.

Reply:
I love how minimal the changes in cranelift are to accommodate this new feature